server/bin/pbench-verify-indexed.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/libexec/platform-python
+# -*- mode: python -*-
+
+"""Pbench Re-Index
+
+Re-index all data for a given date range, "YYYY-MM-DD" to "YYYY-MM-DD".
+
+The process of re-indexing first looks to see which INDEXED, TO-INDEX-TOOL,
+and WONT-INDEX* symlinks exist in the given date range, and moves them to
+TO-RE-INDEX.
+
+NOTE: this interface is intended to be used interactively, this is NOT a
+service that runs as a cronjob.  NO RE-INDEXING STEPS SHOULD BE AUTOMATED
+AT THIS POINT.
+"""
+
+import os
+import re
+import sys
+
+from pathlib import Path
+from argparse import ArgumentParser
+
+from elasticsearch1 import Elasticsearch
+
+import pbench
+
+
+_NAME_ = "pbench-reindex"
+
+tb_pat_r = (
+    r"\S+_(\d\d\d\d)[._-](\d\d)[._-](\d\d)[T_](\d\d)[._:](\d\d)[._:](\d\d)\.tar\.xz"
+)
+tb_pat = re.compile(tb_pat_r)
+
+
+def gen_tb_list(archive):
+    """gen_tb_list - yield all controller/tarball names
+    """
+    with os.scandir(archive) as archive_scan:
+        # We are scanning the archive directory for all controller
+        # sub-directories.
+        for c_entry in archive_scan:
+            if c_entry.name.startswith(".") and c_entry.is_dir(follow_symlinks=False):
+                # Ignore the ".", "..", and any other ".*" subdirectories.
+                continue
```

Comment:
The comment is a bit misleading, since `os.scandir` doesn't return `.` or `..` entries in the first place; but that doesn't really change the code.